validation/schema_testing/officers/officerDeltaSchema_test.go

```diff
@@ -0,0 +1,198 @@
+
+	Convey("Given I want to test the officers-delta API schema to assert ENUM validation is working correctly", t, func() {
+
+		enumErrorsRequestBody := readRequestBody(enumErrorRequestBodyLocation)
+
+		r := httptest.NewRequest("POST", officersEndpoint, bytes.NewBuffer(enumErrorsRequestBody))
+		r.Header.Set("X-Request-Id", contextId)
+		r.Header.Set("Content-Type", "application/json")
+
+		Convey("When I call to validate the request body, providing an valid request with incorrect ENUM values", func() {
+
+			chv := validation.NewCHValidator()
+
+			validationErrs, _ := chv.ValidateRequestAgainstOpenApiSpec(r, apiSpecLocation, contextId)
+
+			Convey("Then I am given an errors array response as validation errors have been found", func() {
+				enumErrorsResponseBody := readRequestBody(enumErrorResponseBodyLocation)
+				match := compareActualToExpected(validationErrs, enumErrorsResponseBody)
+
+				So(validationErrs, ShouldNotBeNil)
+				So(match, ShouldEqual, true)
+			})
+		})
+	})
+}
+
+// readRequestBody takes a file location in the format of a string relative path and reads in the contents from the file,
+// removing any extra tabs, spaces and special characters using json.Compact. Returns a []byte version of the formatted file read in.
+func readRequestBody(fl string) []byte {
+
+	// Read in contents and covert it to a string for further processing.
+	raw, _ := os.ReadFile(fl)
+
+	buffer := new(bytes.Buffer)
+	_ = json.Compact(buffer, raw)
+
+	raw = buffer.Bytes()
+	// Convert back to an []byte and return.
+	return raw
+}
+
+// compareActualToExpected takes an actual and expected byte array and compares them to see if they match. Ordering of response
+// isn't always guaranteed so using this function to match them without having to worry about the order changing.
+func compareActualToExpected(actual, expected []byte) bool {
+
+	// Define 2 model CHError arrays to hold actual and expected responses.
+	var actualErrArr *[]models.CHError
+	var expectedErrArr *[]models.CHError
+
+	// Convert responses to JSON object arrays.
+	_ = json.Unmarshal(actual, &actualErrArr)
+	_ = json.Unmarshal(expected, &expectedErrArr)
+
+	// If the arrays don't match in length, they can't be a complete match so return false.
+	if len(*actualErrArr) != len(*expectedErrArr) {
+		return false
+	}
+
+	// create a map of CHError.Location (string) -> int to compare if both arrays are completely equal.
+	diff := make(map[string]int, len(*expectedErrArr))
+
+	// Range over the expected response array and add them to the newly created map.
+	for _, ee := range *expectedErrArr {
+		// 0 value for int is 0, so just increment a counter for the string
+		diff[ee.Location]++
+	}
+
+	// Finally range over the actual response array errors and check that they exist in the expected errors map.
+	for _, ae := range *actualErrArr {
+		// If the error is not in diff bail out early as they can't be a match.
+		if _, ok := diff[ae.Location]; !ok {
```

Comment:
Could add logging here (to stdout?) to show what is different?

Reply:
I would oppose this only because of the fact that each array will contain close to 44 structs (made of 5 values each) so we would be logging possibly a log of stuff to the console which wouldn't be readable